module/os/linux/zfs/abd_os.c

```diff
@@ -911,11 +911,19 @@ abd_iter_map(struct abd_iter *aiter)
 		aiter->iter_mapsize = aiter->iter_abd->abd_size - offset;
 		paddr = ABD_LINEAR_BUF(aiter->iter_abd);
 	} else {
+		struct page *page;
+		
 		offset = aiter->iter_offset;
 		aiter->iter_mapsize = MIN(aiter->iter_sg->length - offset,
 		    aiter->iter_abd->abd_size - aiter->iter_pos);
 
-		paddr = zfs_kmap_local(sg_page(aiter->iter_sg));
+		page = sg_page(aiter->iter_sg);
+		if (PageHighMem(page)) {
+			page = nth_page(page, offset / PAGE_SIZE);
```

Comment:
Let's change this to `offset >> PAGE_SHIFT` to avoid depending on the compiler to optimize this.